### ISSUE TITLE
Can not publish `datafusion-cli` package to crates.io: expected at most 5 keywords per crate at line 1 column 3441

### ISSUE THREAD

**Issue by alamb** [apache/arrow-datafusion#1332]
**Describe the bug**
While trying to publish the `datafusion-cli` package to crates.io I got the following error:

```shell
(arrow_dev) alamb@MacBook-Pro:~/Downloads/apache-arrow-datafusion-6.0.0$ (cd datafusion-cli && cargo publish)
....
    Finished dev [unoptimized + debuginfo] target(s) in 1m 20s
   Uploading datafusion-cli v5.1.0-SNAPSHOT (/Users/alamb/Downloads/apache-arrow-datafusion-6.0.0/datafusion-cli)
error: failed to publish to registry at https://crates.io

Caused by:
  the remote server responded with an error: invalid upload request: invalid length 7, expected at most 5 keywords per crate at line 1 column 3441
```

**To Reproduce**
```
 (cd datafusion-cli && cargo publish)
```

**Expected behavior**
Upload should work

**Comment by houqp** [comment]
bad news is if we need to release a new version of datafusion cli, we will need to bump datafusion to 7.x because we have breaking change already merged since 6.0.0 :( @alamb @andygrove @Jimexist @Dandandan  what do you think about moving cli also out of the core repo so it can be released at a higher frequency?

Alternatively, we could vote for a datafusion 6.1.0 release, but not actually publishing the core to crates.io to avoid releasing breaking change.

**Comment by alamb** [comment]
> bad news is if we need to release a new version of datafusion cli, we will need to bump datafusion to 7.x because we have breaking change already merged since 6.0.0 :( @alamb @andygrove @Jimexist @Dandandan what do you think about moving cli also out of the core repo so it can be released at a higher frequency?

I think we could make a special `datafusion-cli` release branch (and branch from the 6.0.0 tag) and apply whatever changes were needed rather than having to release whatever is on `master` . 

I find `datafusion-cli` very useful so I would like to keep it in the `arrow-datafusion` repo.

However, we could potentially just release it under a different versioning scheme, which might make sense too 
https://crates.io/search?q=datafusion-cli

**Comment by houqp** [comment]
good idea, a special release branch also works :+1:
